src/Bicep.Cli/Services/CompilationService.cs

```diff
@@ -104,8 +109,27 @@ public ParamsSemanticModel CompileParams(string inputPath, bool skipRestore)
             {
                 throw new Exception($"Unable to read file {inputPath}");
             }
+            
+            var paramsFile = SourceFileFactory.CreateBicepParamFile(inputUri, fileText);
+
+            var usingDeclarations = paramsFile.ProgramSyntax.Children.OfType<UsingDeclarationSyntax>();
+
+            if(!PathHelper.TryGetUsingPath(usingDeclarations.SingleOrDefault(), out var bicepFilePath, out var failureBuilder))
+            {
+                failureBuilder = failureBuilder ?? throw new InvalidOperationException($"{nameof(PathHelper.TryGetUsingPath)} did not provide an error.");
```

Comment:
If you add a `[NotNullWhen(false)]` annotation to the `failureBuilder` out parameter, you could then remove this line.